modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/edit_cp_definition_diagram_setting.js

```diff
@@ -9,15 +9,29 @@
  * distribution rights of the Software.
  */
 
-import {liferayNavigate} from 'commerce-frontend-js/utilities/index';
-import {createPortletURL} from 'frontend-js-web';
+import {fetch} from 'frontend-js-web';
 
-export default function ({currentURL, namespace}) {
-	const portletURL = createPortletURL(currentURL);
+export const HEADERS = new Headers({
+	Accept: 'application/json',
+	'Content-Type': 'application/json',
+});
+
+const DIAGRAMS_ENDPOINT = '/o/headless-commerce-admin-catalog/v1.0/diagrams/';
+
+export default function ({diagramId, namespace}) {
 	const type = document.getElementById(`${namespace}type`);
 	const handleSelectChange = () => {
-		portletURL.searchParams.append('type', type.value);
-		liferayNavigate(portletURL.toString());
+		const url = new URL(
+			DIAGRAMS_ENDPOINT + diagramId,
+			themeDisplay.getPortalURL()
+		);
+		fetch(url, {
+			body: JSON.stringify({
+				type: type.value,
+			}),
+			headers: HEADERS,
+			method: 'PATCH',
+		});
```

Comment:
We should handle errors via `catch` and `!response.ok` so the user can be aware that something is not ok.

(btw I don't see it as blocker in this case)

Reply:
thank you Fabio, will do this after COMMERCE-5254 is merged so I can move this logic in the utilities. js file